### PR TITLE
Setup the teps CI job with git rebase on master

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

The "git batch merge" task in the catalog already does the job of
rebasing (or actually merging) the commits from a PR on top of
master - see https://hub.tekton.dev/tekton/task/git-batch-merge.

Using that task though requires workspaces. I'm using this CI job
then to start the work of removing deprecated resources from our
dogfooding and switching to workspaces and when expressions.

This PR adds a new endpoint for webhooks so that we may
migrate jobs over time from one webhook to the other.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

/kind feature

### CODE REVIEW

**Comment by afrittoli** [comment]
Note that this uses a new ingress, a new event listener and a new folder in git to keep the workspace based CI separated from the pipeline resources based one.

File: tekton/ci-workspace/eventlistener.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            secretRef:
+              secretName: ci-webhook
+              secretKey: secret
+            eventTypes:
+              - pull_request
+        - cel:
+            filter: >-
+              body.repository.full_name.startsWith('afrittoli/') &&
+              body.repository.name in ['tektoncd-community', 'plumbing', 'pipeline', 'triggers', 'cli', 'dashboard', 'catalog', 'hub'] &&
+              (body.action == 'opened' ||
```
Comment (sbwsg) [inline tekton/ci-workspace/eventlistener.yaml:22]: I'm not very familiar with cell but is it possible to use the same construct for `body.action` as is used for `body.repository.name`?  In other words:

```
body.repository.full_name.startsWith('afrittoli/') &&
body.repository.name in ['tektoncd-community', 'plumbing', 'pipeline', 'triggers', 'cli', 'dashboard', 'catalog', 'hub'] &&
body.action in ['opened', 'synchronize', 'labeled', 'unlabeled', 'reopened']
```

The reason I suggest this is to just try and keep the syntax consistent and limited to 3 lines.

File: tekton/ci-workspace/eventlistener.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+               body.action == 'synchronize' ||
+               body.action == 'labeled' ||
+               body.action == 'unlabeled' ||
+               body.action == 'reopened')
+            overlays:
+              - key: git_clone_depth
+                expression: "string(body.pull_request.commits + 1.0)"
+        #     overlays:
```
Comment (sbwsg) [inline tekton/ci-workspace/eventlistener.yaml:30]: nit: suggest dropping commented out code (or leaving a comment explaining when it should be reinstated?)

**Review by sbwsg — COMMENTED** [review]
This looks good to me although I will say that I don't think I'm expert enough to tell whether it's ready to merge and use yet?

File: tekton/ci-workspace/eventlistener.yaml (hunk 1 above)
Comment (afrittoli) [reply tekton/ci-workspace/eventlistener.yaml:22]: Good point, thank you!

File: tekton/ci-workspace/eventlistener.yaml (hunk 2 above)
Comment (afrittoli) [reply tekton/ci-workspace/eventlistener.yaml:30]: will do

File: tekton/ci-workspace/templates/all-template.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          value: check-pr-has-kind-label
+        - name: gitHubCommand
+          value: $(tt.params.gitHubCommand)
+      resources:
+        - name: source
+          resourceSpec:
+            type: git
+            params:
+            - name: revision
+              value: master
```
Comment (vdemeester) [inline tekton/ci-workspace/templates/all-template.yaml:58]: So according to https://github.com/tektoncd/plumbing/issues/681, community will be using `main` starting tomorrow or the day after, and slowly the rest of the projects 🙃

**Comment by afrittoli** [comment]
Uhm, something is not right with the `git-batch-merge` task yet: https://dashboard.dogfooding.tekton.dev/#/namespaces/tekton-ci/pipelineruns/pull-community-teps-lint-5xhsw?pipelineTask=clone-repo&step=clone

**Comment by afrittoli** [comment]
> Uhm, something is not right with the `git-batch-merge` task yet: https://dashboard.dogfooding.tekton.dev/#/namespaces/tekton-ci/pipelineruns/pull-community-teps-lint-5xhsw?pipelineTask=clone-repo&step=clone

Heh, the problem is that the base branch in the `community` repo is `main`, and the task defaults all to `master` :)

**Review by vdemeester — APPROVED** [review]
/meow

**Pull request merged**